lib/modules/dosomething/dosomething_signup/includes/SignupsController.php

```diff
@@ -59,6 +61,37 @@ public function buildContent($entity, $view_mode = 'full', $langcode = NULL, $co
   }
 
   /**
+   * Automagically create a new record.
+   *
+   * @param  array   $values
+   * @return bool|Exception
+   *
+   * @todo   Eventually change this into just a create() method that overrides
+   * the parent EntityAPIController method, and doesn't break dosomething_signup_create()
+   * which does not instantiate the SignupController object, so it has no access to
+   * the $this->save() method below.
+   */
+  public function magick_create(array $values = []) {
+    $signup = parent::create($values);
+
+    // The $this->save() overriden method depends on the use of the
+    // global $user, and then does a check against this user with the user
+    // on the newly created Signup entity. If they don't match then the
+    // signup fails. The ignore_global_user property is set to true so the
+    // afformentioned conditional check on global $user does not happen.
+    $this->ignore_global_user = TRUE;
+
+    try {
+      $record = $this->save($signup);
+
+      // @TODO: eventually return the actual signup record. S'okay for now.
+      return TRUE;
+    } catch (Exception $error) {
+      throw new Exception('There is already an existing signup for this user on this campaign/campaign run combination.');
```

Comment:
We should probably make a `ValidationError` exception class here that we can handle uniquely in the controller. Otherwise any sort of exception (expected or unexpected) would just get reported as a uniqueness thing.